conans/client/graph/range_resolver.py

```diff
@@ -12,18 +12,16 @@ def satisfying(list_versions, versionexpr, output):
     version_range = versionexpr.split(",")
 
     include_prerelease = False
-    if "include_prerelease=True" in version_range:
-        version_range.remove("include_prerelease=True")
-        include_prerelease = True
-    if "include_prerelease=False" in version_range:
-        version_range.remove("include_prerelease=False")
-
     loose = True
-    if "loose=True" in version_range:
-        version_range.remove("loose=True")
-    if "loose=False" in version_range:
-        version_range.remove("loose=False")
-        loose = False
+
+    for keyword in version_range:
```

Comment:
I would prefer something like (not tested):
 
```python
tokens = versionexpr.rsplit(",", 1)
if len(tokens) == 2:
    version_range, operation = tokens
elif len(tokens) == 1:
    version_range, operation = versionexpr, None

include_prerelease = operation == "prerelease"
strict = operation == "strict"
```

Reply:
But we can have 
[>=1.0.0, < 2.0.0, strict, prerelease] or [~1.2.0, prerelease]
So 2-4 tokens.
Anyway, how would we write this range: [1.2.7 || >=1.2.9 <2.0.0], as we normally have to write commas in between?

How about using [>=1.0.0,< 2.0.0; strict; prerelease]? So separating operation with semicolons?

Reply:
Maybe it is worth to use a separator when the operations start. But I'm not sure, to be honest.

Reply:
@conan-io/barbarians feedback needed

Reply:
Any feedback?

Reply:
@lasote: I still wait for some feedback, could you support?

Reply:
@conan-io/barbarians @memsharded @danimtb @jgsogo Feedback.

Reply:
It seems more convenient to have `[<version_range>, <**options>]`, although I think we have to be compatible with existing behavior, which according to the sources is (relevant snippets):

```python
    if version.startswith("[") and version.endswith("]"):
        return version[1:-1]

    version_range = versionexpr.replace(",", " ")
```

Anyway, a simple _regex_ will do the work, something non-greedy like `^\[.+?(,\s*strict)?(,\s*prerrelease)?\]$`, and group matching. I don't feel like the format is so important if `strict` and `prerrelease` are not valid in a version string (maybe we could change them to `loose` and `include_prerrelease` to be safer).

Reply:
Updated to latest comments